docs/src/developing/clients/javascript-reference.md

```diff
@@ -0,0 +1,873 @@
+
+#### Example Usage
+
+```javascript
+const {Authorized, Keypair, Lockup, StakeProgram} = require("@solana/web3.js");
+
+let account = Keypair.generate();
+let stakeAccount = Keypair.generate();
+let authorized = new Authorized(account.publicKey, account.publicKey);
+let lockup = new Lockup(0, 0, account.publicKey);
+
+let createStakeAccountInstruction = StakeProgram.createAccount({
+    fromPubkey: account.publicKey,
+    authorized: authorized,
+    lamports: 1000,
+    lockup: lockup,
+    stakePubkey: stakeAccount.publicKey
+});
+```
+The above code creates a `createStakeAccountInstruction` to be used when creating an account with the `StakeProgram`. The Lockup is set to 0 for both the epoch and Unix timestamp, disabling lockup for the account.
+
+See [StakeProgram](javascript-api.md#StakeProgram) for more.
+
+### Message
+
+[Source Documentation](https://solana-labs.github.io/solana-web3.js/classes/Message.html)
+
+Message is used as another way to construct transactions. You can construct a message using the accounts, header, instructions, and recentBlockhash that are a part of a transaction.
+
+#### Example Usage
+
+```javascript
+const {Buffer} = require("buffer");
+const bs58 = require('bs58');
+const web3 = require('@solana/web3.js');
+
+let toPublicKey = web3.Keypair.generate().publicKey;
+let fromPublicKey = web3.Keypair.generate();
+
+let connection = new web3.Connection(
+    web3.clusterApiUrl('devnet'),
+    'confirmed'
+);
+
+let airdropSignature = await connection.requestAirdrop(
+    fromPublicKey.publicKey,
+    web3.LAMPORTS_PER_SOL,
+);
+
+await connection.confirmTransaction(airdropSignature);
+
+let type = web3.SYSTEM_INSTRUCTION_LAYOUTS.Transfer;
+let data = Buffer.alloc(type.layout.span);
+let layoutFields = Object.assign({instruction: type.index});
+type.layout.encode(layoutFields, data);
+
+let recentBlockhash = await connection.getRecentBlockhash();
+
+let messageParams = {
+    accountKeys: [
+        fromPublicKey.publicKey.toString(),
+        toPublicKey.toString(),
+        web3.SystemProgram.programId.toString()
+    ],
+    header: {
+        numReadonlySignedAccounts: 0,
+        numReadonlyUnsignedAccounts: 1,
+        numRequiredSignatures: 1,
+    },
+    instructions: [
+        {
+        accounts: [0, 1],
+        data: bs58.encode(data),
+        programIdIndex: 2,
+        },
+    ],
+    recentBlockhash,
+};
+
+let message = new web3.Message(messageParams);
+
+let transaction = web3.Transaction.populate(
+    message,
+    [fromPublicKey.publicKey.toString()]
+);
+
+await web3.sendAndConfirmTransaction(connection, transaction, [fromPublicKey])
+```
+
+### NonceAccount
+
+[Source Documentation](https://solana-labs.github.io/solana-web3.js/classes/NonceAccount.html)
+
+Normally a transaction is rejected if a transaction's `recentBlockhash` field is too old. To provide for certain custodial services, Nonce Accounts are used.
```

Comment:
Can you specify that "nonced" transactions (that is, transactions which start with an advance nonce ix) do not expire because the nonce account stores a recent blockhash on-chain which is then specified in the transaction's `recentBlockhash` field?